src/brightness_backend/brightnessctl.rs

```diff
@@ -0,0 +1,162 @@
+use super::{BrightnessBackend, BrightnessBackendConstructor};
+
+const EXPECT_STR: &str = "VirtualDevice didn't test the command during initialization";
+
+use anyhow::bail;
+use thiserror::Error;
+use std::{process::Command, str::FromStr, error::Error};
+
+enum CliArg<'arg> {
+    Simple(&'arg str),
+    KeyValue {
+        key: &'arg str,
+        value: &'arg str,
+    },
+}
+
+impl<'arg> From<&'arg str> for CliArg<'arg> {
+    fn from(value: &'arg str) -> Self {
+        CliArg::Simple(value)
+    }
+}
+
+impl<'arg> From<(&'arg str, &'arg str)> for CliArg<'arg> {
+    fn from((key, value): (&'arg str, &'arg str)) -> Self {
+        CliArg::KeyValue { key, value }
+    }
+}
+
+#[derive(Default)]
+struct VirtualDevice {
+    name: Option<String>,
+    current: Option<u32>,
+    max: Option<u32>,
+}
+
+pub(super) struct BrightnessCtl {
+    device: VirtualDevice,
+}
+
+#[derive(Error, Debug)]
+#[error("Requested device '{device_name}' does not exist ")]
+pub struct DeviceDoesntExistError {
+    device_name: String,
+}
+
+impl VirtualDevice {
+    fn try_new(device_name: Option<String>) -> anyhow::Result<Self> {
+        let s = Self { name: device_name.clone(), ..Default::default() };
+
+        // Check if the command is available to us before running it in other occasions
+        let exit_code = s.command(CliArg::Simple("info"))
+            .output()?
+            .status;
+
+        if exit_code.success() {
+            Ok(s)
+        }
```

Comment:
Any other way of doing this without calling the executable? Does brightnessctl have any APIs that can be used?

Reply:
Not really, [brightnessctl](https://github.com/Hummer12007/brightnessctl) is, in fact, made in a single `.c` file